pkg/function/create_volume_from_snapshot_test.go

```diff
@@ -133,3 +137,51 @@ func (s *CreateVolumeFromSnapshotTestSuite) TestCreateVolumeFromSnapshot(c *C) {
 		}
 	}
 }
+
+func (s *CreateVolumeFromSnapshotTestSuite) TestAddPVProvisionedByAnnotation(c *C) {
+	for _, tc := range []struct {
+		st                  blockstorage.Provider
+		annotations         map[string]string
+		expectedAnnotations map[string]string
+	}{
+		{
+			st:          &gcepd.GpdStorage{},
+			annotations: nil,
+			expectedAnnotations: map[string]string{
+				consts.PVProvisionedByAnnotation: consts.GCEPDProvisionerInTree,
+			},
+		},
+		{
+			st: &gcepd.GpdStorage{},
+			annotations: map[string]string{
+				"key": "value",
+			},
+			expectedAnnotations: map[string]string{
+				"key":                            "value",
+				consts.PVProvisionedByAnnotation: consts.GCEPDProvisionerInTree,
+			},
+		},
+		{
+			st:          &gcepd.GpdStorage{},
+			annotations: map[string]string{},
+			expectedAnnotations: map[string]string{
+				consts.PVProvisionedByAnnotation: consts.GCEPDProvisionerInTree,
+			},
+		},
+		{
+			st: &awsebs.EbsStorage{},
+			annotations: map[string]string{
+				"keyone": "valueone",
+				"keytwo": "valuetwo",
+			},
+			expectedAnnotations: map[string]string{
+				"keyone":                         "valueone",
+				"keytwo":                         "valuetwo",
+				consts.PVProvisionedByAnnotation: consts.AWSEBSProvisionerInTree,
+			},
+		},
+	} {
+		op := addPVProvisionedByAnnotation(tc.annotations, tc.st)
+		c.Assert(op, check.DeepEquals, tc.expectedAnnotations)
```

Comment:
```suggestion
		c.Assert(op, DeepEquals, tc.expectedAnnotations)
```
`gocheck` is imported as `.` above